py/farm_ng/core/events_file_writer.py

```diff
@@ -31,25 +29,22 @@ def is_closed(self) -> bool:
             return True
         return self._file_stream.closed
 
-    def open(self) -> bool:
-        self._file_stream = open(self._file_name, "wb")
+    def open(self, file_name: Path) -> bool:
+        self._file_stream = open(file_name, "wb")
         return self.is_open()
 
     def close(self) -> bool:
-        self._file_stream.close()
-        self._file_stream = None
+        if self._file_stream is not None:
+            self._file_stream.close()
+            self._file_stream = None
         return self.is_closed()
 
-    def write(self, message: Any, uri: Optional[event_pb2.Uri] = None) -> None:
-
-        if uri is None:
-            uri = event_pb2.Uri()
+    def write(self, message: Any, uri: uri_pb2.Uri) -> None:
 
+        # TODO: add timestamps
         event = event_pb2.Event(
-            module=str(message.__module__),
-            name=str(message.__class__.__name__),
-            length_next=message.ByteSize(),
             uri=uri,
+            payload_length=message.ByteSize(),
         )
 
         event_len: bytes = event.ByteSize().to_bytes(1, sys.byteorder)
```

Comment:
I don't think this is a safe assumption that your number of bytes is less than 255 for an Event.

Reply:
4byte maybe?